plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs

```diff
@@ -0,0 +1,119 @@
+                                                 srcSpanEndLine,
+                                                 srcSpanStartCol,
+                                                 srcSpanStartLine, tcg_exports)
+import           Development.IDE.GHC.Error      (realSrcSpanToRange)
+import           GHC.Generics                   (Generic)
+import           HieTypes                       (HieASTs, HieFile)
+import           Ide.Types                      (PluginDescriptor (..),
+                                                 PluginId,
+                                                 defaultPluginDescriptor)
+import qualified Language.LSP.Types             as LSP
+import           Stan.Analysis                  (Analysis (..), runAnalysis)
+import           Stan.Core.Id                   (Id (..))
+import           Stan.Inspection                (Inspection (..))
+import           Stan.Inspection.All            (inspectionsIds, inspectionsMap)
+import           Stan.Observation               (Observation (..))
+
+descriptor :: Recorder (WithPriority Log) -> PluginId -> PluginDescriptor IdeState
+descriptor recorder plId = (defaultPluginDescriptor plId) {pluginRules = rules recorder}
+
+newtype Log = LogShake Shake.Log deriving (Show)
+
+instance Pretty Log where
+  pretty = \case
+    LogShake log -> pretty log
+
+data GetStanDiagnostics = GetStanDiagnostics
+  deriving (Eq, Show, Generic)
+
+instance Hashable GetStanDiagnostics
+
+instance NFData GetStanDiagnostics
+
+type instance RuleResult GetStanDiagnostics = ()
+
+rules :: Recorder (WithPriority Log) -> Rules ()
+rules recorder = do
+  define (cmapWithPrio LogShake recorder) $
+    \GetStanDiagnostics file -> do
+      hie <- fromJust <$> getHieFile file
+      let enabledInspections = HM.fromList [(LSP.fromNormalizedFilePath file, inspectionsIds)]
+      let analysis = runAnalysis Map.empty enabledInspections [] [hie]
+      return (analysisToDiagnostics file analysis, Just ())
+
+  action $ do
+    files <- getFilesOfInterestUntracked
+    void $ uses GetStanDiagnostics $ HM.keys files
+  where
+    analysisToDiagnostics :: NormalizedFilePath -> Analysis -> [FileDiagnostic]
+    analysisToDiagnostics file = map (observationToDianostic file) . toList . analysisObservations
+    observationToDianostic :: NormalizedFilePath -> Observation -> FileDiagnostic
+    observationToDianostic file (Observation {observationSrcSpan, observationInspectionId}) =
+      ( file,
+        ShowDiag,
+        LSP.Diagnostic
+          { _range = realSrcSpanToRange $ observationSrcSpan,
+            _severity = Just LSP.DsHint,
+            _code = Just (LSP.InR $ unId (inspectionId inspection)),
+            _source = Just "stan",
+            _message = message,
+            _relatedInformation = Nothing,
+            _tags = Nothing
+          }
+      )
+      where
+        inspection = fromJust $ HM.lookup observationInspectionId inspectionsMap
```

Comment:
Please don't use partial functions, this can take down HLS entirely. You can log an error and just omit the case you can't handle, just don't `error`.

Reply:
But this function cant fail, because inspection ID always should be from inspection id list.

Reply:
Fixed.